src/hotspot/share/compiler/compilerEvent.cpp

```diff
@@ -101,7 +102,7 @@ int CompilerEvent::PhaseEvent::get_phase_id(const char* phase_name, bool may_exi
     }
 
     index = phase_names->length();
-    phase_names->append(use_strdup ? strdup(phase_name) : phase_name);
+    phase_names->append(use_strdup ? os::strdup(phase_name) : phase_name);
```

Comment:
Where is the `os::free()` to pair with this allocation?

Reply:
AFAICS this is a static global table of phase names that only ever grows and is never deleted.